packages/@storefront/structure/src/carousel/index.ts

```diff
@@ -28,6 +28,9 @@ class Carousel {
 
   onMount() {
     utils.WINDOW().addEventListener('resize', this.forceUpdate);
+
+    // the items have a width of 0 if items are simply set on init, so do an update to recalculate styles
```

Comment:
I understand that you need to force an update, but I don't quite get what you mean by "...if items are simply set on init..."? Can you expand?